src/fast_layers.jl

```diff
@@ -108,34 +115,53 @@ The `activation` function defaults to `identity`, meaning the layer is an affine
 function. Initial parameters are taken to match `Flux.Dense`. The internal
 calculations are done with `StaticArrays` for extra speed for small linear
 algebra operations. Should only be used for input/output sizes of approximately
-16 or less.
+16 or less. 'bias' represents bias(b) in the dense layer and it defaults to true.
 
 Note that this function has specializations on `tanh` for a slightly faster
 adjoint with Zygote.
 """
-struct StaticDense{out,in,F,F2} <: FastLayer
+
+struct StaticDense{out,in,bias,F,F2} <: FastLayer
   σ::F
   initial_params::F2
+  bias::Bool
   function StaticDense(in::Integer, out::Integer, σ = identity;
-                 initW = Flux.glorot_uniform, initb = Flux.zeros)
-    initial_params() = vcat(vec(initW(out, in)),initb(out))
-    new{out,in,typeof(σ),typeof(initial_params)}(σ,initial_params)
+                 bias = true, initW = Flux.glorot_uniform, initb = Flux.zeros)
+    temp = ((bias == true ) ? vcat(vec(initW(out, in)),initb(out)) : vcat(vec(initW(out, in))) )             
+    initial_params() = temp
+    new{out,in,bias,typeof(σ),typeof(initial_params)}(σ,initial_params)
   end
 end
 
 function param2Wb(f::StaticDense{out,in}, p) where {out,in}
-  _W, _b = @views p[1:(out*in)], p[(out*in+1):end]
-  W = @inbounds convert(SMatrix{out,in},_W)
-  b = @inbounds SVector{out}(_b)
+  if f.bias == true
+    _W, _b = @views p[1:(out*in)], p[(out*in+1):end]
+    W = @inbounds convert(SMatrix{out,in},_W)
+    b = @inbounds SVector{out}(_b)
   return W, b
+  else 
+    W = @view p[1:(out*in)]
+    _W = @inbounds convert(SMatrix{out,in},W)
+    return _W

```

Comment:
this now isn't type-stable. It should create `b = SVector{out}()` which is just empty.

Reply:
I didn't quite get your point. Why should we make and return `b = SVector{out}()` if we are not going to use it in the subsequent functions and as far as I know function returns tuple when f.bias == true and it is handled appropriately wherever it is used and same is the case when f.bias == false. The current error is in mnist_gpu.jl which fails even when I am using master repo of DiffEqFlux.jl.

Reply:
It's fixed on master, but that's not the point. The point is speed. This code isn't type-stable because it's dependent on runtime information. The right thing to do might be to make `bias` a type parameter in the case of StaticDense to compensate, or just keep type-stability the easy way.